### PR TITLE
Add airplane runtime env vars to local dev

### PR DESCRIPTION
**Pull request by calufornia**
## Description

Adds most of the environment variables we've documented [here](https://docs.airplane.dev/tasks/runtime-api-reference#environment-variables) to the dev runtime. We omit `AIRPLANE_REQUESTER_EMAIL`, `AIRPLANE_REQUESTER_ID`, `AIRPLANE_SESSION_ID`, `AIRPLANE_TASK_REVISION_ID`, and `AIRPLANE_TRIGGER_ID`.

Depends on https://github.com/airplanedev/lib/pull/283

## Test plan

Created a task the printed out `process.env` and verified the environment variables below were included.

### CODE REVIEW

File: cmd/airplane/tasks/dev/server.go [hunk 1]
```diff
@@ -30,9 +30,22 @@ func runLocalDevServer(ctx context.Context, cfg taskDevConfig) error {
 		TeamID: cfg.root.Client.TeamID,
 	}
 
+	authInfo, err := cfg.root.Client.AuthInfo(ctx)
+	if err != nil {
+		return err
+	}
+
+	var envID string
+	if cfg.envSlug != "" {
```
Comment (colinking) [inline cmd/airplane/tasks/dev/server.go:39]: We'll need to handle the default environment, which is denoted by an empty string for a slug.

File: pkg/dev/dev.go [hunk 2]
```diff
@@ -362,9 +364,31 @@ func entrypointFromDefn(file string) (string, error) {
 
 func appendAirplaneEnvVars(env []string, config LocalRunConfig) ([]string, error) {
 	env = append(env, fmt.Sprintf("AIRPLANE_API_HOST=http://127.0.0.1:%d", config.Port))
-	env = append(env, "AIRPLANE_RUNTIME=dev")
+	env = append(env)
```
Comment (colinking) [inline pkg/dev/dev.go:367]: undo?

File: pkg/dev/dev.go [hunk 3]
```diff
@@ -362,9 +364,31 @@ func entrypointFromDefn(file string) (string, error) {
...
+		runnerID = config.AuthInfo.User.ID
+		runnerEmail = config.AuthInfo.User.Email
+	}
+
+	var teamID string
+	if config.AuthInfo.Team != nil {
+		teamID = config.AuthInfo.Team.ID
+	}
+
+	env = append(env,
```
Comment (colinking) [inline pkg/dev/dev.go:381]: Can you link to the runtime env var docs here? And list out the env vars that we aren't setting here + why, so it's easier to identify if we're missing others.

**Review by colinking — COMMENTED** [review]
(will defer to @kellymtrinh)

File: pkg/dev/dev.go (hunk 2 above)
Comment (calufornia) [reply pkg/dev/dev.go]: Moved with the other env vars below since it's one of the documented environment variables.

**Review by kellymtrinh — APPROVED** [review]
Nice!!!

**Pull request merged**